src/NServiceBus.Core/Routing/FileBasedDynamicRouting/FileRoutingTableFeature.cs

```diff
@@ -33,7 +33,7 @@ public StartupTask(ReadOnlySettings settings, FileRoutingTable routingTable)
 
             protected override Task OnStart(IBusContext contexts)
             {
-                settings.Get<EndpointInstances>().AddDynamic(e => routingTable.GetInstances(e));
+                settings.Get<EndpointInstances>().AddDynamic(e => Task.FromResult(routingTable.GetInstances(e)));
```

Comment:
Nope, because the file is monitored and when the contents change, we want that change be reflected asap, not after the restart of the endpoint. Currently there is no way to remove a statically added instance.